vm/src/builtins.rs

```diff
@@ -724,6 +724,11 @@ fn builtin_zip(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     Ok(vm.ctx.new_list(new_items))
 }
 
+fn builtin_exit(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
```

Comment:
You're going to want to raise the exception with the argument to the exit function as the exception argument. Then, later, convert it to the exit code.